godal.go

```diff
@@ -3849,6 +3849,69 @@ func (ds *Dataset) Grid(destPath string, switches []string, opts ...GridOption)
 	return &Dataset{majorObject{C.GDALMajorObjectH(dsRet)}}, nil
 }
 
+// `Nearblack` and `NearblackInto` run the library version of GDALNearblack. The difference between them is:
+//   - Nearblack: Takes a `destPath` (string) as its first arg, the produced dataset will be output to this "file"
+//     and returned from the function
+//   - NearblackInto: Instead takes a `srcDs` (Dataset) as its first arg, the produced dataset will be stored in
+//     the `Dataset` this method was called from
+//
+// See the nearblack doc page to determine the valid flags/opts that can be set in switches.
+//
+// Example switches :
+//
+//	[]string{"-white", "-near", "10"}
+//
+// Creation options and driver may be set in the switches slice with
+//
+//	switches:=[]string{"-co","TILED=YES","-of","GTiff"}
+//
+// NOTE: Some switches are NOT compatible with this binding, as a `nullptr` is passed to a later call to
+// `GDALNearblackOptionsNew()` (as the 2nd argument). Those switches are: "-o", "-q", "-quiet"
+func (ds *Dataset) Nearblack(destPath string, switches []string, opts ...NearblackOption) (*Dataset, error) {
+	nearBlackOpts := nearBlackOpts{}
+	for _, opt := range opts {
+		opt.setNearblackOpt(&nearBlackOpts)
+	}
+
+	cswitches := sliceToCStringArray(switches)
+	defer cswitches.free()
+
+	dest := unsafe.Pointer(C.CString(destPath))
+	defer C.free(dest)
+
+	cgc := createCGOContext(nil, nearBlackOpts.errorHandler)
+
+	ret, err := C.godalNearblack(cgc.cPointer(), (*C.char)(dest), nil, ds.handle(), cswitches.cPointer())
+	if err = cgc.close(); err != nil {
+		return nil, err
+	}
+
+	return &Dataset{majorObject{C.GDALMajorObjectH(ret)}}, nil
+}
+
+func (ds *Dataset) NearblackInto(srcDs *Dataset, switches []string, opts ...NearblackOption) error {
```

Comment:
this function also needs documenting, by copying/adjusting/referencing the `NearBlack` documentation. Note that godoc documentation *must* start exactly with the the function's name, unquoted, c.f. https://go.dev/blog/godoc